src/source/preprocessor.scala

```diff
@@ -113,7 +113,39 @@ object preprocessor {
   }
 
   private def resolveTypeRef(typeRef: TypeRef, generics: Seq[TypeDecl], specializedGenerics: ArrayBuffer[TypeDecl]): TypeRef = {
-    if (typeRef.expr.args.nonEmpty) {
+    // check whether the type ref is a callback2; if so, we turn it into a Callback or ListCallback; the first
+    // argument, which is all the time "void", is ignored right now
+    if (typeRef.expr.ident.name == "callback2" && typeRef.expr.args.length == 1) {
```

Comment:
Except I misunderstood something I think we have to replace `callback2` by `Callback2`

Reply:
So, not really. The convention is that PascalCase is used by user-defined structures and camelCase for `djinni` native types. Here, `callback2` is a native type I created and injected into `djinni`.